queries/planet_osm_polygon.jinja2

```diff
@@ -0,0 +1,146 @@
+  CASE WHEN mz_transit_level IS NOT NULL
+    THEN jsonb_build_object(
+      'min_zoom', mz_transit_level
+    )
+  END AS __transit_properties__,
+
+  CASE WHEN mz_poi_min_zoom IS NOT NULL
+    THEN jsonb_build_object(
+      'min_zoom', mz_poi_min_zoom,
+      'mz_transit_score', (transit_routes).score,
+      'mz_transit_root_relation_id', (transit_routes).root_relation_id,
+      'train_routes', (transit_routes).train_routes,
+      'subway_routes', (transit_routes).subway_routes,
+      'light_rail_routes', (transit_routes).light_rail_routes,
+      'tram_routes', (transit_routes).tram_routes
+    )
+  END AS __pois_properties__,
+
+  CASE WHEN mz_building_min_zoom IS NOT NULL
+    THEN jsonb_build_object(
+      'min_zoom', mz_building_min_zoom,
+      'name', CASE WHEN mz_poi_min_zoom IS NULL AND mz_landuse_min_zoom IS NULL
+                THEN tags->'name'
+                ELSE tags->'addr:housename'
+              END
+    )
+  END AS __buildings_properties__,
+
+  CASE WHEN mz_water_min_zoom IS NOT NULL
+    THEN jsonb_build_object(
+      'min_zoom', mz_water_min_zoom
+    )
+  END AS __water_properties__,
+
+  CASE WHEN mz_earth_min_zoom IS NOT NULL
+    THEN jsonb_build_object(
+      'min_zoom', mz_earth_min_zoom
+    )
+  END AS __earth_properties__,
+
+  CASE WHEN mz_landuse_min_zoom IS NOT NULL
+    THEN jsonb_build_object(
+      'min_zoom', mz_landuse_min_zoom,
+      'name', CASE WHEN mz_poi_min_zoom IS NULL THEN tags->'name' ELSE NULL END,
+      'mz_is_building', CASE WHEN mz_calculate_is_building_or_part(tags->'building', tags->'building:part') THEN TRUE END
+    )
+  END AS __landuse_properties__
+
+FROM (
+
+  SELECT
+    osm_id,
+    way,
+    mz_label_placement,
+    way_area,
+
+    -- note: the mz_calculate_transit_routes_and_score function is pretty
+    -- expensive, so we only want to calculate it when we actually need the
+    -- result.
+    CASE
+      WHEN mz_poi_min_zoom IS NOT NULL AND
+           tags ? 'railway' AND tags->'railway'='station' AND osm_id > 0
+        THEN mz_calculate_transit_routes_and_score(NULL, osm_id)
+      ELSE NULL
+    END AS transit_routes,
+
+    mz_boundary_min_zoom,
+    mz_building_min_zoom,
+    mz_earth_min_zoom,
+    mz_landuse_min_zoom,
+    mz_poi_min_zoom,
+    mz_transit_level,
+    mz_water_min_zoom,
+
+    tags
+
+  FROM planet_osm_polygon
+
+  WHERE
+
+-- the logic for the filters is slightly different based on whether it's a boundary or not
+{% if zoom >= 16 %}
+    (mz_boundary_min_zoom IS NOT NULL AND {{ bounds['line']|bbox_overlaps('way', 3857) }}) OR
+    ({{ bounds['polygon']|bbox_filter('way', 3857) }} AND
+      (mz_building_min_zoom IS NOT NULL OR
+       mz_earth_min_zoom IS NOT NULL OR
+       mz_landuse_min_zoom IS NOT NULL OR
+       mz_poi_min_zoom IS NOT NULL OR
+       mz_transit_level IS NOT NULL OR
+       mz_water_min_zoom IS NOT NULL))
+{% else %}
+    (mz_boundary_min_zoom < {{ zoom + 1 }} AND {{ bounds['line']|bbox_overlaps('way', 3857) }}) OR
```

Comment:
Suspect we need to have the old **earth**, **road**, **water** layer start_zoom at 8 logic near here now? (Road is probably overkill.)

(Landuse and POI was start 4 but it's fine at 0, should fix in layer min_zoom YAML if an issue.)

(Building and transit is already handled i the layer min_zoom YAML.)